### PR TITLE
[3.5] Split BelongsToMany::sort() into getSort()/setSort()

### PR DESCRIPTION
**Pull request by chinpei215**
`HasMany::sort()` has been split, but `BelongsToMany::sort()` has not been split yet. People would  be confused.

Additionally, I have removed overridden combined setter/getter methods. Previously, `foreignKey()` returns `$this` incorrectly in some of them.

Also, I fixed `BelongsToMany::targetForeignKey()` returns `$this`.

Related to https://github.com/cakephp/cakephp/pull/9676

### CODE REVIEW

**Comment by chinpei215** [comment]
The diff of `BelongsToMany.php` might be confusing a little bit. Please see diffs of each commit.
https://github.com/cakephp/cakephp/pull/10169/commits/7f910a54b84e01915a12ee5f97c77faa72a47d38 https://github.com/cakephp/cakephp/pull/10169/commits/e4978f6f9f4982ef0676aa6ac9683173362c6bf9

File: src/ORM/Association/BelongsTo.php [hunk 1]
```diff
@@ -56,23 +56,6 @@ public function getForeignKey()
...
-     * @return string
-     */
-    public function foreignKey($key = null)
-    {
-        if ($key !== null) {
-            $this->setForeignKey($key);
-        }
-
-        return $this->getForeignKey();
-    }
```
Comment (markstory) [inline src/ORM/Association/BelongsTo.php]: Nice job removing the duplicate code. :+1:

File: tests/TestCase/ORM/Association/HasOneTest.php [hunk 2]
```diff
@@ -79,6 +79,21 @@ public function tearDown()
...
+     * @return void
+     */
+    public function testForeignKey()
+    {
+        $assoc = new HasOne('Profiles', [
+            'sourceTable' => $this->user
+        ]);
+        $this->assertEquals('user_id', $assoc->foreignKey());
+        $this->assertEquals('another_key', $assoc->foreignKey('another_key'));
+        $this->assertEquals('another_key', $assoc->foreignKey());
```
Comment (markstory) [inline tests/TestCase/ORM/Association/HasOneTest.php]: Should there also be tests for the new methods?

Comment (chinpei215) [inline tests/TestCase/ORM/Association/HasOneTest.php]: I think we haven't added any tests for new setters/getters yet.

Comment (markstory) [inline tests/TestCase/ORM/Association/HasOneTest.php]: In some places we haven't. I've been adding them for the request/response classes. But you're right that most of the new ORM methods don't have explicit tests.

Comment (chinpei215) [inline tests/TestCase/ORM/Association/HasOneTest.php]: If they are necessary, I can add tests for new setters/getters of Association classes. Adding all new setters/getters are hard for me 😭

Comment (markstory) [inline tests/TestCase/ORM/Association/HasOneTest.php]: Don't worry about it then. We can do it later when we add runtime errors to the deprecated methods.

Comment (chinpei215) [inline tests/TestCase/ORM/Association/HasOneTest.php]: OK.

**Comment by chinpei215** [comment]
I noticed a fact that I have to wait for 3.5 if I get this PR merged 😞

**Comment by chinpei215** [comment]
Hmm. I have to separate this PR, as I want to fix a bug introduced in #9676. That should be done in mater, not 3.next.

**Pull request closed without merging**